### PR TITLE
Scaleset/runnerjob scalesetclient

### PR DESCRIPTION
**Pull request by fhammerl**

### CODE REVIEW

File: pkg/github/runnermanager/clientset.go [hunk 1]
```diff
@@ -23,11 +25,21 @@ func CreateJob(ctx context.Context, jitConfig *github.RunnerScaleSetJitRunnerCon
...
-	createdJob, err := clientset.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+	job := runnerJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace) // defaultJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace)
+
+	body, err := json.Marshal(job)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal job")
+	}
+
+	_, err = clientset.RESTClient().
+		Post().
```
Comment (fhammerl) [inline pkg/github/runnermanager/clientset.go:44]: Pass in the ctx for cancellation

File: pkg/github/runnermanager/clientset.go [hunk 2]
```diff
@@ -23,11 +25,21 @@ func CreateJob(ctx context.Context, jitConfig *github.RunnerScaleSetJitRunnerCon
 		return nil, err
 	}
 
-	job := defaultJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace)
-	createdJob, err := clientset.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+	job := runnerJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace) // defaultJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace)
```
Comment (fhammerl) [inline pkg/github/runnermanager/clientset.go]: Clean up the comment

File: pkg/github/runnermanager/clientset.go [hunk 3]
```diff
@@ -23,11 +25,21 @@ func CreateJob(ctx context.Context, jitConfig *github.RunnerScaleSetJitRunnerCon
...
+	body, err := json.Marshal(job)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal job")
+	}
+
+	_, err = clientset.RESTClient().
+		Post().
+		AbsPath(fmt.Sprintf("/apis/actions.summerwind.dev/v1alpha1/namespaces/%s/runnerjobs", namespace)).
+		Body(body).
+		DoRaw(context.TODO())
```
Comment (fhammerl) [inline pkg/github/runnermanager/clientset.go]: What is this line?

Comment (fhammerl) [reply pkg/github/runnermanager/clientset.go]: Read runnerjob and return that object

File: pkg/github/runnermanager/clientset.go [hunk 4]
```diff
@@ -23,11 +25,30 @@ func CreateJob(ctx context.Context, jitConfig *github.RunnerScaleSetJitRunnerCon
 		return nil, err
 	}
 
-	job := defaultJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace)
-	createdJob, err := clientset.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+	job := runnerJobResource(jitConfig.EncodedJITConfig, jitConfig.Runner.Id, namespace)
```
Comment (fhammerl) [inline pkg/github/runnermanager/clientset.go]: `runnerJobResource` -> `runnerJobTemplate`?

**Review by nikola-jokic — APPROVED** [review]
LGTM

**Pull request merged**